src/conversions/mod.rs

```diff
@@ -1,5 +1,6 @@
 //! This module contains conversions between non-String Rust object and their string representation
 //! in Python
 
+pub mod array;
```

Comment:
I think there is no need for this to be `pub` if we move all code which is concerned with arrays into it.